add Employee interface

zbychhhu opened the pull request:
Employee interface with getters

zbychhhu committed:
add Employee interface

czechadrian commented:
możemy skorzystać z Lomboka który generuje gettery,settery i inne boilerplatowe metody. Dodanie zależności,zainstalowanie pluginu w Intellj i przełączenie coś tam w ustawieniach zapewnia używanie go.

zbychhhu committed:
add lombok and AbstractEmployee class

zbychhhu reviewed (commented):
dodałem tego lomboka ,czy tak to ma wyglądać ?

MPomaranski committed:
Creating base code, composite, employees, work calendar, car.

MPomaranski committed:
Package name from Car into car.

czechadrian committed:
feature/AGH_7
remove previous classes and replace them with generic class which could be use in every type of employee class.

type of storing workcalendar is temporary - it is possible to be change to ArrayList<T> workCalendar or to String == if in the next tasks it is necessary

czechadrian commented:
takie rozwiązanie jest lepsze ,bo oszczędzamy dobre kilka klas mając to w jednej... tak jak w opisie commita podałem - pole workCalendar może się zmienić z tej listy w zależności od tego jak będzie najwygodniej z save/findBy*/findAll itd. Sprawdzcie ,czy jest ok i dajcie znać czy macie jakiś inny pomysł.

zbychhhu commented:
jak dla mnie spoko ,niech leci merge
